src/main/java/strategy/SupplyOperationHandler.java

```diff
@@ -0,0 +1,10 @@
+package strategy;
+
+import java.util.Map;
+
+public class SupplyOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> inventory, String fruit, int quantity) {
+        inventory.put(fruit, inventory.getOrDefault(fruit, 0) + quantity);
```

Comment:
According to the checklist, the inventory should be encapsulated and not directly manipulated. Consider refactoring the code to follow the encapsulation principle.